services/graphql-server/src/graphql/resolvers/email/schedule.js

```diff
@@ -15,6 +17,57 @@ const clearSeconds = (date) => {
 };
 
 module.exports = {
+  /**
+   *
+   */
+  Query: {
+
+    emailSchedulesForDateRange: async (_, { input }, { basedb, site }) => {
+      const {
+        newsletterId,
+        beforeEndOf,
+        startingStartOf,
+      } = input;
+
+      // Use input timezone otherwise fallback to site/global timezone.
+      const timezone = input.timezone || site.get('date.timezone', defaults.date.timezone);
+      const startDate = momentTZ(startingStartOf).tz(timezone);
+      const endDate = momentTZ(beforeEndOf).tz(timezone);
+      const start = startDate.startOf('day').toDate();
+      const end = endDate.endOf('day').toDate();
+
+      const scheduleSort = { sequence: 1, deploymentDate: 1 };
+
+      const scheduleQuery = {
+        product: BaseDB.coerceID(newsletterId),
+        status: 1,
+        deploymentDate: { $gte: start, $lte: end },
+      };
+      const schedules = await basedb.find('email.Schedule', scheduleQuery, {
+        sort: scheduleSort,
+        projection: {
+          _id: 1,
+          product: 1,
+          'content.$id': 1,
+          section: 1,
+          deploymentDate: 1,
+          sequence: 1,
+          status: 1,
+        },
```

Comment:
The 4th argument to a GraphQL resolver is the [info](https://www.apollographql.com/docs/apollo-server/data/resolvers/#resolver-arguments) param, which includes metadata about the requested fields at query time.

You should build this value dynamically based on the fields the user requested. There's no need to include `section` or `product` (and incur the db query to materialize their data) if the user only requested the `content` and `id` fields).

You can build this data automatically using the [buildProjection](https://github.com/parameter1/base-cms/blob/master/services/graphql-server/src/graphql/utils/build-projection.js) utility:
```js
emailSchedulesForDateRange: async (root, vars, context, info) => {
  // ..
  const projection = buildProjection({ info, type: 'EmailSchedule' });
```

Reply:
Thanks for pointing this out, I had only seen `getProjection` within `newsletterScheduledContent` and wasn't sure how to get a valid projection for this. This seems to have worked as I needed it to here.